zebra-state/src/service/finalized_state/disk_db.rs

```diff
@@ -512,6 +512,38 @@ impl DiskWriteBatch {
 }
 
 impl DiskDb {
+    /// Prints rocksdb metrics for each column family along with total database disk size, live data disk size and database memory size.
+    pub fn print_db_metrics(&self) {
+        let mut total_size_on_disk = 0;
+        let mut total_size_in_mem = 0;
+        let mut total_live_size_on_disk = 0;
+        let db: &Arc<DB> = &self.db;
+        let db_options = DiskDb::options(); 
+
+        let column_families = DiskDb::construct_column_families(&db_options, db.path(), &[]);
+    
+        for cf_descriptor in column_families {
+            let cf_name = &cf_descriptor.name();
+            let cf_handle = db.cf_handle(cf_name).expect("Column family handle must exist");
+    
+            let live_data_size = db.property_int_value_cf(cf_handle, "rocksdb.estimate-live-data-size").unwrap_or(Some(0));
+            let total_sst_files_size = db.property_int_value_cf(cf_handle, "rocksdb.total-sst-files-size").unwrap_or(Some(0));
+    
+            let cf_disk_size = live_data_size.unwrap_or(0) + total_sst_files_size.unwrap_or(0);
+            total_size_on_disk += cf_disk_size;
+            total_live_size_on_disk += live_data_size.unwrap_or(0);
+    
+            let mem_table_size = db.property_int_value_cf(cf_handle, "rocksdb.size-all-mem-tables").unwrap_or(Some(0));
+            total_size_in_mem += mem_table_size.unwrap_or(0);
+    
+            info!("Column Family: {}, Disk Size: {} bytes, Memory Size: {} bytes", cf_name, cf_disk_size, mem_table_size.unwrap_or(0));
+        }
+    
+        info!("Total Database Disk Size: {} bytes", total_size_on_disk);
+        info!("Total Live Data Disk Size: {} bytes", total_live_size_on_disk);
+        info!("Total Database Memory Size: {} bytes", total_size_in_mem);
```

Comment:
I think the totals are the information most users will want. here is an example of that:

```
2024-03-11T23:09:18.536494Z  INFO {zebrad="1cf4e6c" net="Main"}: zebra_state::service::finalized_state::disk_db: Total Database Disk Size: 36483859660 bytes
2024-03-11T23:09:18.536524Z  INFO {zebrad="1cf4e6c" net="Main"}: zebra_state::service::finalized_state::disk_db: Total Live Data Disk Size: 18019914845 bytes
2024-03-11T23:09:18.536552Z  INFO {zebrad="1cf4e6c" net="Main"}: zebra_state::service::finalized_state::disk_db: Total Database Memory Size: 49152 bytes
```
Do you think it will be possible to display the bytes in megs or gigs (human readable form)? I understand this might need a dependency for example https://crates.io/crates/human_bytes so we might want to leave it for later and add a `TODO` comment into the code.

By the way, do you think the size in mem a useful metric ?

Reply:
@oxarbitrage All checks resolved